app/controllers/Accounts.scala

```diff
@@ -86,14 +86,17 @@ class Accounts(implicit val env: GlobalEnvironment)
 
   def passwordResetCommit = SecuredAction { implicit request =>
     ResetPasswordForm.passwordsForm.bindFromRequest.fold(
-      formWithErrors => Ok(views.html.passwordreset(formWithErrors)),
+      formWithErrors => {
```

Comment:
There are four conditions that can be wrong:
- a required field is empty
- original password is wrong
- the new password is not long enough (<6)
- the new passwords do not match.
Currently the message displayed will basically always be "New passwords don't match". We should probably  check the conditions named in the order listed above and the appropriate messages forwarded to the user.

Reply:
The "original password is wrong" error is already displayed correctly.